File: challenges/contests/code_forces/round326/c.cpp
/*
 * c.cpp
 * Copyright (C) 2016 missingdays <missingdays@missingdays>
 *
 * Distributed under terms of the MIT license.
 */

#include <iostream>

using namespace std;

const int MAX = 1000100;

int a[MAX];

int main(){
    ios::sync_with_stdio(false);

    int n;
    cin >> n;

    for(int i = 0; i < n; i++){
        int x;
        cin >> x;

        a[x]++;
    }

    int answ = 0;

    for(int i = 0; i < MAX-1; i++){
        a[i+1] += a[i]/2;

        a[i] %= 2;

        answ += a[i];
    }

    cout << answ << endl;
    return 0;
}

File: challenges/contests/code_forces/round326/c.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# vim:fenc=utf-8
#
# Copyright © 2016 missingdays <missingdays@missingdays>
#
# Distributed under terms of the MIT license.

"""

"""

MAXN = int(1e6) + 100

n = int(input())
m = [0 for i in range(MAXN)]
answ = 0

a = list(map(int, input().split()))

for e in a:
    m[e] += 1

for i in range(MAXN-1):
    m[i+1] += m[i]//2

    m[i] %= 2

    answ += m[i]

print(answ)
